src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_helpers.cc

```diff
@@ -276,4 +276,86 @@ void RCHelpers::RemoveRedundantGPSDataFromIVDataMsg(
     location_data = new_location_data;
   }
 }
+
+smart_objects::SmartObject RCHelpers::MergeModuleData(
+    const smart_objects::SmartObject& data1,
+    const smart_objects::SmartObject& data2) {
+  if (data1.getType() != smart_objects::SmartType::SmartType_Map ||
+      data2.getType() != smart_objects::SmartType::SmartType_Map) {
+    return data2;
+  }
+
+  smart_objects::SmartObject result = data1;
+
+  for (auto it = data2.map_begin(); it != data2.map_end(); ++it) {
+    const std::string& key = it->first;
+    smart_objects::SmartObject& value = it->second;
+    if (!result.keyExists(key) || value.getType() != result[key].getType()) {
+      result[key] = value;
+      continue;
+    }
+
+    // Merge maps and arrays with `id` param included, replace other types
+    if (value.getType() == smart_objects::SmartType::SmartType_Map) {
+      value = MergeModuleData(result[key], value);
+    } else if (value.getType() == smart_objects::SmartType::SmartType_Array) {
+      value = MergeArray(result[key], value);
+    }
+    result[key] = value;
+  }
+  return result;
+}
+
+smart_objects::SmartObject RCHelpers::MergeArray(
+    const smart_objects::SmartObject& data1,
+    const smart_objects::SmartObject& data2) {
+  // Merge data only in the case where each value in the array is an Object with
+  // an ID included, otherwise replace
+
+  if (data1.getType() != smart_objects::SmartType::SmartType_Array ||
+      data2.getType() != smart_objects::SmartType::SmartType_Array ||
+      data2.empty()) {
+    return data2;
+  }
+
+  auto& data2_array = *data2.asArray();
+  for (const auto& e : data2_array) {
```

Comment:
@YarikMamykin Bad variable name even it is temporary. `e` is a bad name. NAme it at least `item`

Reply:
@AByzhynar , fixed in 66a90ade0c